O/OpenFOAM_com/build_tarballs.jl

```diff
@@ -0,0 +1,576 @@
+    ExecutableProduct("surfaceGenerateBoundingBox", :surfaceGenerateBoundingBox),
+    ExecutableProduct("surfaceHookUp", :surfaceHookUp),
+    ExecutableProduct("surfaceInertia", :surfaceInertia),
+    ExecutableProduct("surfaceInflate", :surfaceInflate),
+    ExecutableProduct("surfaceLambdaMuSmooth", :surfaceLambdaMuSmooth),
+    ExecutableProduct("surfaceMeshConvert", :surfaceMeshConvert),
+    ExecutableProduct("surfaceMeshExport", :surfaceMeshExport),
+    ExecutableProduct("surfaceMeshExtract", :surfaceMeshExtract),
+    ExecutableProduct("surfaceMeshImport", :surfaceMeshImport),
+    ExecutableProduct("surfaceMeshInfo", :surfaceMeshInfo),
+    ExecutableProduct("surfaceOrient", :surfaceOrient),
+    ExecutableProduct("surfacePatch", :surfacePatch),
+    ExecutableProduct("surfacePointMerge", :surfacePointMerge),
+    ExecutableProduct("surfaceRedistributePar", :surfaceRedistributePar),
+    ExecutableProduct("surfaceRefineRedGreen", :surfaceRefineRedGreen),
+    ExecutableProduct("surfaceSplitByPatch", :surfaceSplitByPatch),
+    ExecutableProduct("surfaceSplitByTopology", :surfaceSplitByTopology),
+    ExecutableProduct("surfaceSplitNonManifolds", :surfaceSplitNonManifolds),
+    ExecutableProduct("surfaceSubset", :surfaceSubset),
+    ExecutableProduct("surfaceToFMS", :surfaceToFMS),
+    ExecutableProduct("surfaceToPatch", :surfaceToPatch),
+    ExecutableProduct("surfaceTransformPoints", :surfaceTransformPoints),
+    ExecutableProduct("surfactantFoam", :surfactantFoam),
+    ExecutableProduct("temporalInterpolate", :temporalInterpolate),
+    ExecutableProduct("tetgenToFoam", :tetgenToFoam),
+    ExecutableProduct("tetMesh", :tetMesh),
+    ExecutableProduct("thermoFoam", :thermoFoam),
+    ExecutableProduct("topoSet", :topoSet),
+    ExecutableProduct("transformPoints", :transformPoints),
+    ExecutableProduct("twoLiquidMixingFoam", :twoLiquidMixingFoam),
+    ExecutableProduct("twoPhaseEulerFoam", :twoPhaseEulerFoam),
+    ExecutableProduct("uncoupledKinematicParcelDyMFoam", :uncoupledKinematicParcelDyMFoam),
+    ExecutableProduct("uncoupledKinematicParcelFoam", :uncoupledKinematicParcelFoam),
+    ExecutableProduct("viewFactorsGen", :viewFactorsGen),
+    ExecutableProduct("vtkUnstructuredToFoam", :vtkUnstructuredToFoam),
+    ExecutableProduct("wallFunctionTable", :wallFunctionTable),
+    ExecutableProduct("writeMeshObj", :writeMeshObj),
+    ExecutableProduct("writeMorpherCPs", :writeMorpherCPs),
+    ExecutableProduct("XiDyMFoam", :XiDyMFoam),
+    ExecutableProduct("XiEngineFoam", :XiEngineFoam),
+    ExecutableProduct("XiFoam", :XiFoam),
+    ExecutableProduct("zipUpMesh", :zipUpMesh),
+    
+    FileProduct("share/openfoam/etc", :openfoam_etc), 
+    FileProduct("share/openfoam/bin", :openfoam_bin)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency(PackageSpec(name="flex_jll", uuid="48a596b8-cc7a-5e48-b182-65f75e8595d0"))
```

Comment:
To reiterate the question in https://github.com/JuliaPackaging/Yggdrasil/pull/7976#discussion_r1478728642, how's flex used?  Is it needed at runtime?  Or only at build time?  If you mark it as `BuildDependency` and then it's needed at runtime, no one will tell you until you need it and things crash badly.